File: weather-library/src/main/java/org/rcgonzalezf/weather/common/models/converter/ModelConverter.java
package org.rcgonzalezf.weather.common.models.converter;

import java.io.IOException;
import java.util.List;

public interface ModelConverter<D extends Data, E, W extends Data> {

  void fromForecastPojo(D pojo);

  void fromWeatherPojo(W pojo);

  List<E> getForecastModel() throws IOException;

  List<E> getWeatherModel() throws IOException;
}

File: weather-library/src/main/java/org/rcgonzalezf/weather/common/models/converter/ModelConverter.kt
package org.rcgonzalezf.weather.common.models.converter

import java.io.IOException

interface ModelConverter<D : Data?, E, W : Data?> {
    fun fromForecastPojo(pojo: D)
    fun fromWeatherPojo(pojo: W)

    @get:Throws(IOException::class)
    val forecastModel: List<E>?

    @get:Throws(IOException::class)
    val weatherModel: List<E>?
}
